sample/src/main/java/com/karumi/dexter/sample/SampleApplication.java

```diff
@@ -27,5 +29,14 @@
   @Override public void onCreate() {
     super.onCreate();
     Dexter.initialize(this);
+    boolean launchDexterOnApplication = DexterApplicationMode.isEnable(this);
+    if (launchDexterOnApplication) {
+      DexterApplicationMode.clear(this);
+      requestDexterOnApplication();
+    }
+  }
+
+  private void requestDexterOnApplication() {
+    Dexter.checkPermission(new EmptyPermissionListener(), Manifest.permission.RECORD_AUDIO);
```

Comment:
I'd use one with the `token.continuePermissionRequest();` method called in the `onPermissionRationaleShouldBeShown` callback.